BACK-447 - Add No milestone option to TUI milestone filters

MrLesk opened the pull request:
## Summary
- Treats the old TASK-24.1 PR as the idea source and rebuilds the branch on current main.
- Adds the Web UI No milestone option to the existing TUI milestone filters for task list and Kanban views.
- Uses a shared milestone filter sentinel so task list and board filtering stay consistent.

## Related Task
BACK-447 - Add No milestone option to TUI milestone filters

## Testing
- bun test src/test/unified-view-filters.test.ts src/test/task-viewer-milestone-filter-model.test.ts src/test/filter-header-navigation.test.ts src/test/cli-milestone-filter.test.ts
- bunx tsc --noEmit
- bun run check .
- bun test

orakemu commented:
Looking forward to this one!
What does your current mental roadmap/priorities look like?

MrLesk committed:
BACK-447 - Add No milestone TUI filter option

MrLesk commented:
Alex's Agent: I rebuilt this branch on current `main` as `BACK-447 - Add No milestone option to TUI milestone filters`. The original `TASK-24.1` diff was treated as an idea only; this now extends the existing TUI milestone filter UX with Web-parity `No milestone` support for both task list and Kanban. Local verification passed (`bun test`, `bunx tsc --noEmit`, `bun run check .`). @codex review

MrLesk commented:
Alex's Agent: Fixed the Codex review finding by replacing the user-visible `__none` sentinel with an internal NUL-prefixed TUI control value, and added a regression test proving a real milestone titled `__none` still filters normally. Local focused tests, typecheck, and Biome passed. @codex review
